mailbox/api/src/main/java/org/apache/james/mailbox/extractor/TextExtractor.java

```diff
@@ -23,8 +23,14 @@
 
 import org.apache.james.mailbox.model.ContentType;
 
+import reactor.core.publisher.Mono;
+
 public interface TextExtractor {
 
     ParsedContent extractContent(InputStream inputStream, ContentType contentType) throws Exception;
 
+    default Mono<ParsedContent> extractContentReactive(InputStream inputStream, ContentType contentType) {
+        return Mono.fromCallable(() -> extractContent(inputStream, contentType));
```

Comment:
given that extractContent is blocking, would it makes sense to enforce this to be executed in an elastic scheduler?

Reply:
Good question. In practice this is, with current implementation, a byte array input stream. But yes I will put that subscribeOn for the sake of genericity or we might regret it later.

Good catch!